kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/write/row/BigQueryWriter.java

```diff
@@ -102,15 +106,25 @@ public void writeRows(PartitionedTableId table,
     logger.debug("writing {} row{} to table {}", rows.size(), rows.size() != 1 ? "s" : "", table);
 
     Exception mostRecentException = null;
+    Map<Long, List<BigQueryError>> partialFailRowsMap = null;
 
     int retryCount = 0;
     do {
       if (retryCount > 0) {
         waitRandomTime();
       }
       try {
-        performWriteRequest(table, rows, topic);
-        return;
+        partialFailRowsMap = performWriteRequest(table, rows, topic);
+        if (partialFailRowsMap == null || partialFailRowsMap.isEmpty()) {
+          // table insertion completed with no reported errors
+          return;
+        } else {
+          // update insert rows based on failed rows in case of partial failure
+          // use partialFailRowsMap to track error instead of mostRecentException
+          rows = getFailRows(rows, partialFailRowsMap.keySet());
```

Comment:
lets have a logging message in here saying that so many rows succeeded and so many rows failed for table foo, and that we are retrying. Like the logging messages when we get a BigQueryException.

Reply:
sure we can do that